### PR TITLE
Replace getRewindCapabilities with feature checks

### PR DESCRIPTION
**Pull request by obenland**
#### Changes proposed in this Pull Request

* Updates remaining `getRewindCapabilities` calls with feature checks.
* Removes `getRewindCapabilities` selector.
* Removes unused `getAllowRestore` selector.

#### Testing instructions

* Navigate to https://container-frosty-leakey.calypso.live/backup/ and select a site
* With a site that has Realtime Backups (eCommerce) make sure the detail view is being shown: image
* With a site on a lower tier plan, check that upgrade buttons are still shown.
* [Create a new jurassic.ninja site](https://jurassic.ninja/specialops/) with a backup subscription.
* Once connected, try purchasing Jetpack Security and make sure the warning banner shows: Screen Shot 2022-05-24 at 10 14 42 AM

Related to p4TIVU-a66-p2

### CODE REVIEW

File: client/state/selectors/get-allow-restore.js [hunk 1]
```diff
@@ -22,12 +22,12 @@ export default createSelector(
...
 			'active' === rewind.state &&
 			! ( 'queued' === restoreStatus || 'running' === restoreStatus ) &&
-			includes( siteCapabilities, 'restore' )
+			siteHasRestore
 		);
 	},
 	( state, siteId ) => {
 		return [
-			getRewindCapabilities( state, siteId ),
+			siteHasFeature( state, siteId, WPCOM_FEATURES_BACKUPS_RESTORE ),
```
Comment (obenland) [inline client/state/selectors/get-allow-restore.js]: Am I doing this right? I don't have much experience with `createSelector()`

File: client/my-sites/backup/hooks.js [hunk 2]
```diff
@@ -61,15 +61,9 @@ export const useFirstMatchingBackupAttempt = (
 	siteId,
 	{ before, after, successOnly, sortOrder } = {}
 ) => {
-	const dispatch = useDispatch();
-
-	useEffect( () => {
-		dispatch( requestRewindCapabilities( siteId ) );
-	}, [ siteId ] );
```
Comment (obenland) [inline client/my-sites/backup/hooks.js:68]: @jsnajdr I initially tried to replace `requestRewindCapabilities` with `fetchSiteFeatures ` but it caused an infinite render loop when visiting https://container-elastic-elgamal.calypso.live/backup with a WoA site so I removed that in 98775b5f7f6f1dd745ff72301f6571a2b9d839d4. 

I saw [you recently worked in this area](https://github.com/Automattic/wp-calypso/pull/60756) and was curious if you could help me figure out if removing this would be okay or how I could rewrite it to avoid re-rendering?

File: client/state/selectors/get-allow-restore.js (hunk 1 above)
Comment (jsnajdr) [reply client/state/selectors/get-allow-restore.js]: I think this selector doesn't need to use `createSelector` at all. And the original version didn't need it either.

There is no complex calculation going on, just looking up properties on objects. And the return value is a `boolean` primitive, not any complex object or array.

Typically you need `createSelector` memoization to ensure that calling the selector with same inputs (state and params) returns the same output. Where "same" means `===` to each other.

Selector like this does it automatically because it returns a `boolean`:
```js
const hasBigNumbers = ( state, limit ) => state.numbers.some( n => n > limit );
```
but this one returns a new array each time even if `state` and `limit` are identical:
```js
const getBigNumbers = ( state, limit ) => state.numbers.filter( n => n > limit );
```
It's the latter type where `createSelector` is useful to limit rerenders.

File: client/state/sites/products/conflicts.ts [hunk 3]
```diff
@@ -165,7 +165,8 @@ export const isPlanIncludingSiteBackup = createSelector(
 		( state: AppState, siteId: number | null, planSlug: string ) => [
 			siteId,
 			planSlug,
-			getRewindCapabilities( state, siteId ),
+			siteHasFeature( state, siteId, WPCOM_FEATURES_BACKUPS ),
+			siteHasFeature( state, siteId, WPCOM_FEATURES_REAL_TIME_BACKUPS ),
```
Comment (jsnajdr) [inline client/state/sites/products/conflicts.ts:169]: None of the selectors in this module need the `createSelector` optimization. The only complex computation that's going on is the `getAllFeaturesForPlan` function that constructs a new array on each call. This could be optimized in a separate PR.

File: client/my-sites/backup/hooks.js (hunk 2 above)
Comment (jsnajdr) [reply client/my-sites/backup/hooks.js:68]: This hook (`useFirstMatchingBackupAttempt`) is used by the `RealtimeStatus` component and the fetch happens when that component mounts.

Now `RealtimeStatus` is rendered by `BackupStatus`, conditionally when site features have finished fetching. The code is like:
```js
function BackupStatus() {
  if ( isFetchingSiteFeatures ) {
    return <BackupPlaceholder />;
  }
  return <RealtimeStatus />;
}
```
And the infinite loop goes like this:
1. Someone (e.g. the top level `Layout` with `QuerySiteFeatures`) triggers a site features fetch.
2. `BackupStatus` renders a placeholder.
3. Site features fetch finishes.
4. `BackupStatus` renders `RealtimeStatus` instead of the placeholder.
5. That means that `RealtimeStatus` _mounts_ and on mount it triggers a site features fetch again.
6. `BackupStatus` renders the placeholder again because `isFetchingSiteFeatures` is true again.
7. Site features fetch finishes and `RealtimeStatus` is mounted again...

We need to be more careful with `fetchSiteFeatures` and `QuerySiteFeatures`. The features are always queried in the top-level `Layout` component, so no one else should need to query them explicitly.

Ideally, the fetch would happen only when i) we're not already fetching and ii) the fetch hasn't already finished. In other words, if site features are already in state, cache them forever and don't even re-fetch them.

Comment (obenland) [reply client/my-sites/backup/hooks.js:68]: Thanks, in that case I'll leave it removed

File: client/state/selectors/get-allow-restore.js (hunk 1 above)
Comment (obenland) [reply client/state/selectors/get-allow-restore.js]: It was introduced in #42302 and doesn't seem to be used anymore. I've not been able to pinpoint where usage was removed though.

It's probably safe to delete the selector alltogether.

File: client/state/sites/products/conflicts.ts (hunk 3 above)
Comment (obenland) [reply client/state/sites/products/conflicts.ts:169]: Yes, happy to give it a shot when we'll refactor the other plan checks in here

File: client/state/selectors/get-allow-restore.js [hunk 4]
```diff
@@ -1,35 +0,0 @@
-import { createSelector } from '@automattic/state-utils';
```
Comment (mreishus) [inline client/state/selectors/get-allow-restore.js:1]: Last reference to get-allow-restore removed in #54973

Comment (obenland) [reply client/state/selectors/get-allow-restore.js:1]: Thanks! I was not able to track that down.

**Review by mreishus — APPROVED** [review]
LGTM, I traced all of the changed selectors, and they're working properly for sites with and without backup, including switching between sites, fetching features and purchase conflicts.

**Review by jsnajdr — APPROVED** [review]
Looks good to me now :+1:
